solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java

```diff
@@ -634,17 +635,19 @@ private static long getSegmentsFileLength(IndexCommit commit) {
 
   /** Returns the sum of RAM bytes used by each segment */
   private static long getIndexHeapUsed(DirectoryReader reader) {
-    long indexHeapRamBytesUsed = 0;
-    for(LeafReaderContext leafReaderContext : reader.leaves()) {
-      LeafReader leafReader = leafReaderContext.reader();
-      if (leafReader instanceof SegmentReader) {
-        indexHeapRamBytesUsed += ((SegmentReader) leafReader).ramBytesUsed();
-      } else {
-        // Not supported for any reader that is not a SegmentReader
-        return -1;
-      }
-    }
-    return indexHeapRamBytesUsed;
+    return reader.leaves().stream()
+        .map(LeafReaderContext::reader)
+        .map(FilterLeafReader::unwrap)
+        .map(leafReader -> {
+          if (leafReader instanceof Accountable) {
```

Comment:
can we filter here instead?

Reply:
or map into accountable returning -1L

Reply:
Feel free to suggest a code snippet; I'm not sure I get it.

One small improvement I see I could make was change the last "map" call to be a "mapToLong" instead, and then I can remove the last mapToLong.